src/CMakeLists.txt

```diff
@@ -695,6 +695,11 @@ endif()
 include(CheckCSourceCompiles)
 include(CheckSymbolExists)
 
+check_symbol_exists(strlcpy "string.h" HAVE_STRLCPY)
+if(HAVE_STRLCPY)
+	add_definitions(-DHAVE_STRLCPY)
```

Comment:
Please use cmake_config.h like here:
https://github.com/minetest/minetest/blob/a8cf10b0b523bf4c3234be29b0dd51e719cb00c1/src/cmake_config.h.in#L32

Reply:
done, thanks!

Reply:
I'm afraid this approach has downside in CMake always setting `HAVE_STRLCPY`. I'll test if `#if HAVE_STRLCPY == 0` could work correctly

Reply:
should be resolved by https://github.com/minetest/minetest/pull/14279 sadly I can test ppc64le and s390x only in the CI, so I didn't catch it right away.